### PR TITLE
Add on-demand CI trigger

### PR DESCRIPTION
**Pull request by calebmshafer**
The current triggers for the CI build don't include the ability to manually queue it if needed.

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -1,6 +1,7 @@
 name: CI Build
 
 on:
+  workflow_dispatch:
```
Comment (jffmarker) [inline .github/workflows/ci.yaml:4]: Are you sure you want the trailing colon?

Comment (jffmarker) [reply .github/workflows/ci.yaml:4]: I tried it and I think you actually do...

Comment (calebmshafer) [reply .github/workflows/ci.yaml:4]: Yea, I thought it was strange as well but this is how I saw it used in their docs.

**Comment by calebmshafer** [comment]
@williamkbentley mind approving?

**Pull request merged**